torch/autograd/functional.py

```diff
@@ -429,6 +436,42 @@ def jacobian(func, inputs, create_graph=False, strict=False):
     _check_requires_grad(outputs, "outputs", strict=strict)
 
     jacobian: Tuple[torch.Tensor, ...] = tuple()
+
+    if vectorize:
+        for outidx, out in enumerate(outputs):
```

Comment:
contrary to what is said in the docstring, you actually call autograd.grad once per output here.
Is that expected?

Reply:
Can we actually do that in a single call by giving a tuple of output/grad_output?

Reply:
> contrary to what is said in the docstring, you actually call autograd.grad once per output here.
Is that expected?

For the docstring, I meant to say "... calls autograd.grad once per output".

> Can we actually do that in a single call by giving a tuple of output/grad_output?

I never thought about this before. It turns out that we can! (and probably should). Let me think about it for a bit and then update the PR

Reply:
So the answer is yes, we can and should just do one call. However, this comes with the following caveat:
- `vectorize=True` and `strict=True` doesn't quite work. This is because instead of returning `None` for autograd.grad, `vectorize=True` might cause it to return a tensor full of zeros.
- I am thinking of just raising an error message if the above two are passed in together.

Reply:
I guess an error is ok for now.
But in the future, I think it would be ok to get just 0s for these and not raise the errror?

Reply:
> But in the future, I think it would be ok to get just 0s for these and not raise the errror?

Is it OK for `strict=False` to cause different behavior when `vectorize` is True and when it is False? I thought that would have been strange

Reply:
You mean `strict=True` right?
Yes I agree with you that it is surprising and forbidding it is definitely the best move for now.

Reply:
Yes, I meant `strict=True`, my bad. That sounds good (y)